javaScript-week2/candy-helper.js

```diff
@@ -0,0 +1,36 @@
+const boughtCandyPrices = [];
+
+const candyPrices = {
+  "sweet": 0.5,
+  "chocolate": 0.7,
+  "toffee": 1.1,
+  "chewing-gum": 0.03,
+};
+
+function addCandy(candyType, weight) {
+  const totalCost = candyPrices[candyType] * weight;
+  boughtCandyPrices.push(totalCost);
+}
+
+let amountToSpend = Math.random() * 100;
+
+function canBuyMoreCandy() {
+  let totalSpent = 0;
+  for (let i = 0; i < boughtCandyPrices.length; i++) {
+    totalSpent += boughtCandyPrices[i];
+  }
+
```

Comment:
The code has an issue where you never get a different result than "Enough candy for you". To debug it, you could add here the following logs:
```
  console.log("totalSpent: ", totalSpent);
  console.log("amountToSpend: ", amountToSpend);
```

What this would show you is that `totalSpent: NaN`. This is because you call `addCandy` once without an argument for `type`. May be you can write logic that checks for this case and returns a message as "Sorry, but we don't have that type of candy", or something else.